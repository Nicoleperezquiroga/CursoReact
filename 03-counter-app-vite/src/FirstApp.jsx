import PropTypes from 'prop-types';

const newMessage = 'Nicole';
//Para poner un objeto en una etiqueta
//Se debe usar JSON.stringify(newMessage)
const newMessage1 = {
    message: 'Hola mundo',
    title: 'Fernando'
}

const getMensaje = () => {
    return 'FunciÃ³n getMensaje';
}
export const FirstApp = ({title, subtitle, name }) =>{
    return (
    <>
        {/*SOY UN COMENTARIO*/}
        {/*<h1>{title}</h1>*/}
        
        <div data-testid="test-title">{title}</div>
        <h1>{title}</h1>

        <p>{subtitle}</p>
         {/*<p>{subtitle}</p>*/}
        <h1>{getMensaje()}</h1>
        <code>{ JSON.stringify(newMessage1) }</code>
        <h1>{newMessage}</h1>
        <p>{name}</p>
    </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subtitle: 123,
    name: 'Fernando'
}