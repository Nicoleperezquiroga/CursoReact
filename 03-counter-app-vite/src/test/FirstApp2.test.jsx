import { render, screen } from "@testing-library/react";
import { FirstApp } from "../FirstApp";
//snapshot se asegura que la componente no sufra cambios
describe('Pruebas en <FirstApp/>', () => {
    const title = 'Hola, Soy Goku';
    const subtitle = 'Soy un subtitulo';
    test('debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe hacer el mensaje Hola, Soy Goku', () => {
        render(<FirstApp title={title}/>);
        expect(screen.getAllByText(title)).toBeTruthy();
    });

    test('debe mostrar un titulo h1', () => {
        render(<FirstApp title={title}/>);
        console.log(screen.getAllByRole('heading',{level:1}).innerHTML)

        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('debe mostrar un subtitulo enviado por props', () => {
        render(<FirstApp title={title} subtitle={subtitle}/>);
        expect(screen.getAllByText(subtitle).length).toBe(1);    });
});
