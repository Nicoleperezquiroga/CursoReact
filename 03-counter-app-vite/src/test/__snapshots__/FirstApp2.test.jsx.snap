// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, Soy Goku
  </div>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    123
  </p>
  <p>
    Funci√≥n getMensaje
  </p>
  <p>
    {"message":"Hola mundo","title":"Fernando"}
  </p>
  <p>
    Nicole
  </p>
  <p>
    Fernando
  </p>
</div>
`;
